purchases/src/test/java/com/revenuecat/purchases/backend_integration_tests/GoldenFileRecorder.kt

```diff
@@ -0,0 +1,253 @@
+    }
+
+    private fun filterHeaders(headers: Map<String, String>): Map<String, String> {
+        return headers.filterKeys { key ->
+            // Keep the header only if it doesn't match any of the ignore patterns
+            !headerIgnorePatterns.any { pattern -> pattern.matches(key) }
+        }
+    }
+
+    private fun filterBody(body: String?): String? {
+        if (body == null || bodyFieldsToIgnore.isEmpty()) {
+            return body
+        }
+
+        return try {
+            val jsonObject = JSONObject(body)
+            bodyFieldsToIgnore.forEach { pathSpec ->
+                applyFieldFilter(jsonObject, pathSpec)
+            }
+            jsonObject.toString()
+        } catch (@Suppress("SwallowedException") _: Exception) {
+            // Not a JSON object or parsing failed, return as-is
+            body
+        }
+    }
+
+    /**
+     * Applies filtering to a field specified by a path.
+     * Path syntax:
+     * - "field" - top-level field
+     * - "parent > child" - nested field
+     * - "parent > \[field\]" - field in each element of parent array
+     */
+    private fun applyFieldFilter(jsonObject: JSONObject, pathSpec: String) {
+        val parts = pathSpec.split(">").map { it.trim() }
+        navigateAndFilter(jsonObject, parts, 0)
+    }
+
+    private fun navigateAndFilter(current: Any, parts: List<String>, index: Int) {
+        if (index >= parts.size) return
+
+        val part = parts[index]
+        val isLast = index == parts.size - 1
+
+        // Check if this part is an array field accessor like "[fieldName]"
+        if (part.startsWith("[") && part.endsWith("]")) {
+            val fieldName = part.substring(1, part.length - 1)
+            if (current is JSONArray) {
+                // Apply to each element in the array
+                for (i in 0 until current.length()) {
+                    val element = current.get(i)
+                    if (element is JSONObject && element.has(fieldName)) {
+                        if (isLast) {
+                            element.put(fieldName, "TEST_STATIC_VALUE")
+                        } else {
+                            navigateAndFilter(element.get(fieldName), parts, index + 1)
+                        }
+                    }
+                }
+            }
+        } else {
+            // Regular field access
+            if (current is JSONObject && current.has(part)) {
+                if (isLast) {
+                    current.put(part, "TEST_STATIC_VALUE")
+                } else {
+                    navigateAndFilter(current.get(part), parts, index + 1)
+                }
+            }
+        }
+    }
+
+    private fun recordGoldenFile(sequenceNumber: String, request: RecordedRequest, response: RecordedResponse) {
+        val requestFile = File(testDirectory, "request_$sequenceNumber.json")
+        val responseFile = File(testDirectory, "response_$sequenceNumber.json")
```

Comment:
I see requests/responses are being stored as `request_001`, `request_002`.

I'm wondering if we should be so strict about checking the requests (we'd be also checking the requests order). My thinking is that this could break fairly easily if we add new requests, change the requests orders, etc. I'm not sure anyway